### ISSUE TITLE
Why use socket to connect to servers and run commands instead of SSH?

### ISSUE THREAD

**Issue by wydwww** [douban/tfmesos#62]
According to my understanding, in tfmesos, `tfrun` connects to `server.py` on the remote nodes through socket, then `server.py` runs the TensorFlow script received from scheduler by subprocess.

Why not run subprocess to SSH login to the servers then run the TF scripts? I know that Spark adopts SSH to login to workers, and SSH is more convenient and safer than socket.

Thanks!

**Comment by windreamer** [comment]
Because we are using Mesos to manage resources of the whole cluster.
With Mesos, multiple training can be parallel and independent with each other. and the resources are dynamically and fairly allocated for each task.

So this project is mainly to bridge the gap of Tensorflow and Mesos

**Comment by wydwww** [comment]
@windreamer Yes I understand tfmesos connects TF and Mesos. I mean the scheduler makes connections with nodes in the cluster through socket, then send the hyperparameters and cmd to every node, so the node can run tf training code.

I guess it's better for the scheduler to `subprocess.check_call('ssh node_addr & cmd')`? It can avoid socket connection and server.py running on every node.

**Comment by windreamer** [comment]
Because for Mesos, task should be launched by Mesos executor instead of ssh or anything else. This is the requirement of Mesos for isolate and manage the task.

**Comment by wydwww** [comment]
@windreamer Thanks!
